Add cfn template for bastion

niroy opened the pull request:
This template launches a bastion host on customer vpc.
Instance has Mate AMI so you ssh and rdp both to it.

Attachment to transit gateway and tg route updates is currently automated, may have to be removed later so that learner manually does it .

We may automate adding the oc binary to the bastion host in future.

Everything has been tested so far and is working fine.

niroy committed:
Add cfn template for bastion

This template launches a bastion host on customer vpc
Instance has Mate AMI so you ssh and rdp both to it
I have automated the attachment to transit gateway and tg route updates, that part may have to be removed later so that learner manually does it
We may automate adding the oc binary to the bastion host in future

herve4m commented on aws/cloudformation/rdp_bastion.yaml:
```diff
@@ -0,0 +1,492 @@
+---
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >
+  This template creates a bastion host based on Mate AMI on a new VPC(customer).
+  This bastion host will be used to connect to the ROSA cluster, run oc commands
+  and view the OpenShift console.
```
Missing space before the parenthesis:

```suggestion
  This template creates a bastion host based on Mate AMI on a new VPC
  (customer).
  This bastion host will be used to connect to the ROSA cluster, run oc commands
  and view the OpenShift console.
```

herve4m commented on aws/cloudformation/rdp_bastion.yaml:
```diff
@@ -0,0 +1,492 @@
+        To: 443
+  rInboundNetworkACLRule4:
+    Type: AWS::EC2::NetworkAclEntry
+    Properties:
+      NetworkAclId:
+        Fn::GetAtt:
+          - rCustomerVpc
+          - DefaultNetworkAcl
+      RuleNumber: 53
+      Protocol: 6
+      RuleAction: allow
+      CidrBlock: 0.0.0.0/0
+      PortRange:
+        From: 22
+        To: 22
+  rInboundNetworkACLRule5:
+    Type: AWS::EC2::NetworkAclEntry
+    Properties:
+      NetworkAclId:
+        Fn::GetAtt:
+          - rCustomerVpc
+          - DefaultNetworkAcl
+      RuleNumber: 54
+      Protocol: 6
+      RuleAction: allow
+      CidrBlock: 0.0.0.0/0
+      PortRange:
+        From: 3389
+        To: 3389
+  rInboundNetworkACLRule6:
+    Type: AWS::EC2::NetworkAclEntry
+    Properties:
+      NetworkAclId:
+        Fn::GetAtt:
+          - rCustomerVpc
+          - DefaultNetworkAcl
+      RuleNumber: 60
+      Protocol: -1
+      RuleAction: deny
+      CidrBlock: 0.0.0.0/0
```
Those default network ACL were for the ROSA cluster. A customer VPC would not have those. Maybe remove all those ACLs and keep the defaults (allow everything), or deny everything (as it was the initial plan: https://drive.google.com/file/d/1PmRi_iKmiVR4pnHxJFJtEs-wEkURNRat/view?usp=sharing) and then add network ACL at the subnet level for just what you need.

niroy replied:
Thanks, will change the nacl rules accordingly in the next commit.

niroy committed:
Fix tg with id <> not found

TG needs to get attached to new vpc before route can be created else error like this:

The transitGateway ID 'tgw-0a1eb8e73fb6aec5d' does not exist

niroy committed:
Fix spacing

Missing space before parenthesis

Co-authored-by: Hervé Quatremain <herve4m@users.noreply.github.com>

niroy committed:
Update nacl rules and remove tg automation

1. Change default vpc nacl to deny all inbound
2. Create nacls with rules for private & public subnet
3. Remove section for transit gateway as learner will do it manually

antmaro commented on aws/cloudformation/rdp_bastion.yaml:
```diff
@@ -0,0 +1,501 @@
+  pTagprefix:
+    Description: Use the course code to prepend resource tags
+    Type: String
+    Default: cs220
```
```suggestion
    Default: cs220-cluster
```

Review the naming to fit the Herve's Cloudformation style

niroy replied:
The customer vpc is outside the cs220-cluster (rosa) that's why I chose `cs220` and not `cs220-cluster`

niroy replied:
@herve4m and @antmaro can I keep the tag prefix as cs220? else I'll have to update it everywhere in the commands while retrieving the IDs of the resources.

niroy replied:
I'm keeping it as cs220 as it's easier to distinguish the resources of customer-vpc in AWS console

antmaro commented on aws/cloudformation/rdp_bastion.yaml:
```diff
@@ -0,0 +1,501 @@
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pTagprefix, "-customer-vpc"]]
```
```suggestion
          Value: !Join ["", [!Ref pTagprefix, "-customer-vpc-bastion"]]
```

Review the changes from our conversation regarding bastion/vpn

antmaro reviewed (changes requested):
As per our conversation about having two Customer VPCs VPN/Bastion, review my suggestions ;-)

niroy committed:
Update tag customer-vpc to customer-vpc-bastion

Co-authored-by: Antonio Marí Romero <amarirom@redhat.com>

niroy committed:
Add cs220 tag and fix spaces
